### PR TITLE
remove unused patch; update test command and requirements

### LINKED ISSUE #25: The tests may no longer be run on the feedstock in version 0.4.0.0
Issue:

The Slycot devs reworked the tests and they may not be running anymore with this command:

```
python -c "from slycot import test; test()"
```

See: https://github.com/python-control/Slycot/issues/98#issuecomment-636377862

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list

```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info

```
</details>

### PR DESCRIPTION
**Pull request by bnavigator**
As discussed in https://github.com/python-control/Slycot/issues/98#issuecomment-636379272

I am not very familiar with the conda-forge and smithy workflow, but let's try:

Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] @conda-forge-admin, please rerender
* [ ] Ensured the license file is being packaged.

Fixes #25

### CODE REVIEW

**Comment by bnavigator** [comment]
[Windows, Python 3.8](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=168750&view=logs&j=e5cdccbf-4751-5a24-7406-185c9d30d021&t=66e4a587-a723-53e9-873c-6c9a2c5971ea&l=3461):
```
================================== FAILURES ===================================
____________________________ test_xerbla_override _____________________________

    def test_xerbla_override():
        """Test that Fortran routines calling XERBLA do not print to stdout."""
    
        try:
            out = subprocess.check_output([sys.executable, '-c', CODE],
                                           stderr=subprocess.STDOUT,
                                           universal_newlines=True)
        except subprocess.CalledProcessError as cpe:
            raise RuntimeError("Trying to call _wrapper.ab08nd() failed with "
                               "returncode {}.\n"
                               "Captured STDOUT: \n {}\n"
                               "Captured STDERR: \n {}\n"
                               "".format(cpe.returncode, cpe.stdout, cpe.stderr))
    
        outlines = out.splitlines()
        assert len(outlines) == 2
>       assert outlines[0] == _wrapper.__file__
E       AssertionError: assert 'D:\\bld\\sly...win_amd64.pyd' == 'D:\\bld\\sly...win_amd64.pyd'
E         Skipping 42 identical trailing characters in diff, use -v to show
E         - %PREFIX%\Lib\site-p
E         ?                                       ^
E         + %PREFIX%\lib\site-p
E         ?                                       ^
```

Welcome to the world of case-insensitive path names :facepalm: 

Going to fix this by skipping the test until we fix it in the Slycot source.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -31,11 +31,13 @@ requirements:
 
 test:
   requires:
-    - nose
+    - pytest
+    - scipy
   imports:
     - slycot
   commands:
-    - python -c "from slycot import test; test()"
```
Comment (moorepants) [inline recipe/meta.yaml:38]: So, does this command not work anymore?

**Comment by moorepants** [comment]
LGMT!

File: recipe/meta.yaml (hunk 1 above)
Comment (bnavigator) [reply recipe/meta.yaml:38]: Seems so. I did not notice that before. The command imports Tester from numpy which uses nose...

Off to file a bug in the Slycot repository and working on a PR.

Comment (bnavigator) [reply recipe/meta.yaml:38]: Done in https://github.com/python-control/Slycot/pull/136

Found https://github.com/numpy/numpy/pull/10827, which means numpy did the same transition and newer numpy.Tester would call pytest again.

**Pull request merged**